File: Data Structures & Algorithms/binary-search/submission-2.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:

        #Simple binary Search
        prev_values = set()
        def BS(start, end):
            
            if (start > end):
                return -1

            index = int((start + end)/2)

            if (nums[index] == target):
                return index

            if (nums[index] > target):
                return BS(start, index-1)
            else:
                return BS(index+1, len(nums)-1)

        #I like to start at the middle
        return BS(0,len(nums)-1)

File: Data Structures & Algorithms/sort-colors/submission-2.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        vector<int> freq(3, 0); for (int n: nums) { freq[n]++; }
        int index = 0;
        for (int i = 0; i < freq.size(); i++) {
            int n = freq[i];
            for (int j = 0; j < n; j++) {
                nums[index] = i;
                index++;
            }
        }
    }
};